rust2/core/src/options.rs

```diff
@@ -0,0 +1,500 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Various option and configuration types.
+use std::os::raw::c_int;
+
+use crate::{
+    error::{Error, Status},
+    ffi::constants,
+};
+
+/// Option value.
+///
+/// Can be created with various implementations of [From].
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum OptionValue {
+    String(String),
+    Bytes(Vec<u8>),
+    Int(i64),
+    Double(f64),
```

Comment:
I would prefer to keep `Double` because that's what the C spec uses. 

Go is in a similar situation with a "double" type named `float64` and it uses `SetOptionDouble(key string, value float64) error`.